File: tarea-3/etapa-4/game_src/Inventario.cpp
#include <iostream>
#include <string>
#include <vector>
#include "../game_headers/Item.h"
#include "../game_headers/Inventario.h"

Inventario::Inventario(Mascota &mascota) : mascota(mascota) 
{}

Inventario::~Inventario() {
    for (Item *item : items) {
		delete item;
	}
}

void Inventario::agregarItem(Item *item) {
    items.push_back(item);
}

Item * Inventario::buscarPorID(int id) {
	for (Item *item : items) {
		if (item->getId() == id)
		    return item;
	}
	return NULL;
}

void Inventario::usarItem(int id, Mascota &mascota) {
	Item *item = buscarPorID(id);
    item->usar(mascota);

	int index = 0;
	for (Item *item : items) {
		if (item->getId() == id)
            break;
		index++;
	}

	if (item->getCantidad() == 0) {
		items.erase(items.begin() + index);
	}
}

std::vector<Item*> Inventario::getItems() {
    return items;
}

void Inventario::mostrarInventario() const{
	for (Item *item : items) {
		item->mostrarItem();
	}
};

File: proyecto/inventario.py
from Item import Item
from Mascota import Mascota
from PyQt6.QtCore import *
from PyQt6.QtGui import *
from PyQt6.QtSql import *
from PyQt6.QtWidgets import *

class inventario(QLabel):
    def __init__(self, mascota):
         self.items = []
         self.mascota = mascota
    def agregar_item(self, item):
        self.items.append(item)
        

    def buscar_por_id(self, id):
        for item in self.items:
            if item.id == id:
                return item
        return None

    def usar_item(self, id, mascota):
        item = self.buscar_por_id(id)
        if item:
            item.usar(mascota)
            self.items.remove(item)

    def get_items(self):
        return self.items

    def mostrar_inventario(self):
        if not self.items:
            print("El inventario está vacío.")
        else:
            for item in self.items:
                print(item)